File: First UI/app/src/main/java/com/example/firstui/MainActivity.java
package com.example.firstui;

import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import android.os.Handler;
import android.widget.*;
import android.view.View;

public class MainActivity extends AppCompatActivity {
    ProgressBar progressBar;
    Handler handler = new Handler();
    int progressStatus = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        EditText editTextName = findViewById(R.id.nameEditText);
        Button buttonGreet = findViewById(R.id.greetButton);
        TextView textViewGreeting = findViewById(R.id.greetingTextView);
//
        CheckBox checkBoxSubscribe = findViewById(R.id.subscribeCheckBox);
        RadioGroup radioGroupColors = findViewById(R.id.colorRadioGroup);
        Switch switchNotifications = findViewById(R.id.notificationSwitch);
//
        SeekBar seekBarVolume = findViewById(R.id.volumeSeekBar);
        TextView textViewVolume = findViewById(R.id.seekBarLabel);
//
        Spinner spinnerCountry = findViewById(R.id.countrySpinner);
        Button buttonSubmit = findViewById(R.id.submitButton);
        progressBar = findViewById(R.id.loadingProgressBar);

        // Setup Spinner (Countries)
        String[] countries = {"USA", "Canada", "Germany", "India", "UK"};
        ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.simple_spinner_item, countries);
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        spinnerCountry.setAdapter(adapter);

        // Greet Button
        buttonGreet.setOnClickListener(v -> {
            String name = editTextName.getText().toString().trim();
            String greeting = "Hello, " + name + "!";
            textViewGreeting.setText(greeting);
        });

        // SeekBar Listener
        seekBarVolume.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                textViewVolume.setText("Volume: " + progress);
            }

            @Override public void onStartTrackingTouch(SeekBar seekBar) {}
            @Override public void onStopTrackingTouch(SeekBar seekBar) {}
        });

        // Simulate loading on button click
        buttonSubmit.setOnClickListener(v -> simulateLoading(switchNotifications.isChecked(),checkBoxSubscribe.isChecked()));
    }

    private void simulateLoading(boolean notifications,boolean subscribed) {
        progressBar.setVisibility(View.VISIBLE);
        progressStatus = 0;
        progressBar.setProgress(0);

        new Thread(() -> {
            while (progressStatus < 100) {
                progressStatus += 5;
                handler.post(() -> progressBar.setProgress(progressStatus));
                try {
                    Thread.sleep(100); // 100ms * 20 = 2 seconds
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
            }
            handler.post(() -> progressBar.setVisibility(View.GONE));
        }).start();
        if(notifications){
          if(subscribed){
              Toast.makeText(this,
                      "Subscribed ✅",
                      Toast.LENGTH_LONG).show();
          }
          else {
              Toast.makeText(this,
                      "Not Subscribed ❌",
                      Toast.LENGTH_LONG).show();
          }
        }
    }
}

File: first_ui_kotlin/app/src/main/java/com/example/first_ui_kotlin/MainActivity.kt
package com.example.first_ui_kotlin

import android.os.Bundle
import android.view.View
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.CheckBox
import android.widget.EditText
import android.widget.ImageView
import android.widget.ProgressBar
import android.widget.RadioGroup
import android.widget.SeekBar
import android.widget.SeekBar.OnSeekBarChangeListener
import android.widget.Spinner
import android.widget.Switch
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions

class MainActivity : AppCompatActivity() {

    private lateinit var loadingPB : ProgressBar
    private var progressStetus : Int = 0


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main) // <-- ADD THIS LINE
        enableEdgeToEdge()

        loadingPB = findViewById(R.id.loadingProgressBar)
        var greetingTextView : TextView = findViewById(R.id.greetingTextView)
        var nameEditText : EditText = findViewById(R.id.nameEditText)
        var greetButton : Button = findViewById(R.id.greetButton)
        var imageView : ImageView = findViewById(R.id.imageView)

        var subscribeCheckBox : CheckBox = findViewById(R.id.subscribeCheckBox)
        var notificationSwitch : Switch = findViewById(R.id.notificationSwitch)
        var favoratColorRGroup : RadioGroup = findViewById(R.id.colorRadioGroup)
        var seekBarLabel : TextView = findViewById(R.id.seekBarLabel)
        var seekBarVolume : SeekBar = findViewById(R.id.volumeSeekBar)
        var countrySpinner : Spinner = findViewById(R.id.countrySpinner)

        var submitButton : Button = findViewById(R.id.submitButton)


        // Setup Spinner (Countries)
        val countries = arrayOf("India","Canada","Germany","USA","UK")
        val adapter = ArrayAdapter(this,android.R.layout.simple_spinner_item,countries)
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        countrySpinner.setAdapter(adapter)

        // Greet Button
        greetButton.setOnClickListener(View.OnClickListener { v: View? ->
            val name: String = nameEditText.getText().toString().trim()
            val greeting = "Hello, $name!"
            greetingTextView.setText(greeting)

            nameEditText.visibility = View.GONE
            greetButton.visibility = View.GONE

            Glide.with(this)
                .load(R.drawable.wellcome)
                .transition(DrawableTransitionOptions.withCrossFade(1000))
                .into(imageView)
        })


        // SeekBar Listener
        seekBarVolume.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                seekBarLabel.setText("Volume: $progress")
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {}
            override fun onStopTrackingTouch(seekBar: SeekBar) {}
        })

    }
}
